scripts/scil_visualize_fodf.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Visualize 2-dimensional fODF slice loaded from disk.
+
+Given a SH coefficients image, this script displays a slice in the
+orientation specified by the user. The user can also add a background
+on top of which the fODF are to be displayed.
+"""
+
+import argparse
+
+import nibabel as nib
+import numpy as np
+
+from dipy.data import get_sphere
+
+from scilpy.io.utils import (add_sh_basis_args, add_overwrite_arg,
+                             assert_inputs_exist, assert_outputs_exist)
+from scilpy.io.image import get_data_as_mask
+from scilpy.viz.scene_utils import (create_odf_slicer, create_texture_slicer,
+                                    create_peaks_slicer, create_scene,
+                                    render_scene)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    # Positional arguments
+    p.add_argument('in_fodf', default=None, help='Input SH image file.')
+
+    p.add_argument('--slice_index', type=int,
+                   help='Index of the slice to visualize along a given axis.')
+
+    # Window configuration options
+    p.add_argument('--win_dims', nargs=2, metavar=('WIDTH', 'HEIGHT'),
+                   default=[768, 768], type=int,
+                   help='The dimensions for the vtk window.')
+
+    p.add_argument('--interactor', default='trackball',
+                   choices={'image', 'trackball'},
+                   help='Specify interactor mode for vtk window.')
+
+    p.add_argument('--axis_name', default='axial', type=str,
+                   choices={'axial', 'coronal', 'sagittal'},
+                   help='Name of the axis to visualize.')
+
+    p.add_argument('--silent', action='store_true',
+                   help='Disable interactive visualization.')
+
+    p.add_argument('--output', help='Path to output file.')
+
+    add_overwrite_arg(p)
+
+    # Optional FODF personalization arguments
+    p.add_argument('--sh_order', type=int, default=8,
```

Comment:
This should be replaced by `sh_order = order_from_ncoef(data['fodf'].shape[-1])`  later in your code. The function is from `dipy.reconst.shm`.